### PR TITLE
SPP-10421-update-design-system

### PR DESCRIPTION
**Pull request by joepercival**
# Description

Following an accessibility audit on SML the Design System team have made a number of changes to fix the issue identified. An update to the latest version is required in order for SML to actually benefit from that work. This pull request will update the design system, and make any adjustments due to breaking changes.

Documentation has not been updated and comments have not been provided since they are not applicable for this pull request.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Technical Enhancements

# Checklist:

## **Definition of Done**
**Code and merges**

- [ ] Code to be commented on where applicable 
- [ ] Documentation updated where required 
- [x] Have considered non-functional requirements such as Security, Performance, Scalability, and Fault Tolerance
- [x] I have linted the code

**Testing**

- [x] All levels of acceptance test are passing (automated, integration, manual, accessibility, etc.)
- [x] I have run the behave command to check the selenium behaviour tests pass locally
- [x] Acceptance criteria met

**Other Checks**
- [x] I have performed a self-review of my own code
- [x] I have checked for spelling errors
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes don't break anything unexpected
- [x] I have checked and updated the security.txt file where required
- [x] Up to date with the main branch
- [x] For a change that needs to be deployed in a release I have a commit with the fix:, feat: or BREAKING CHANGE:
tags and will copy this commit message to the squash
- [ ] For a change that does not affect the deployed code I have added a commit with ci:, docs: or test:
depending upon which area of the solution was affected

Note when merging, squash commit must begin with one of the following tags:
"BREAKING CHANGE", "feat", "fix", "ci", "docs", "test"

Usage

BREAKING CHANGE: any change that breaks current functionality and may cause compatibility errors to users upgrading
    E.G BREAKING CHANGE: refactored method x to take variable y as input (changed from variable z)

feat: A new feature
    E.G feat: adding method x to future methods table

fix: A bug fix
    E.G fix: amending page for method x to show correct release version

ci: Changes to our CI configuration files and scripts (Concourse)

docs: Documentation only changes

test: Adding missing tests or correcting existing tests

### CODE REVIEW

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d16l1gzj7635sf.cloudfront.net/
cloudfront_id=E3BVER8ZW4V1OV

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d16l1gzj7635sf.cloudfront.net/
cloudfront_id=E3BVER8ZW4V1OV

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d16l1gzj7635sf.cloudfront.net/
cloudfront_id=E3BVER8ZW4V1OV

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d16l1gzj7635sf.cloudfront.net/
cloudfront_id=E3BVER8ZW4V1OV

**Review by jhmjam — DISMISSED** [review]

**Comment by gibbardsteve** [comment]
@joepercival I have not reviewed the code yet but I did do a little testing first to see if things are behaving ok.  I found the following, could you take a look please?

**Cookies banner:**
Clicking on “additional cookies” or “view cookies” is not loading the cookies page.

**Cookies link in footer:**
Clicking this just flicks the page to the header rather than show me the cookies? Note: it only exhibits this behaviour when the cookies banner is present.

**Home**
“Useful resources” is a link now that returns 404. On the deployed website this isn’t a link.

**Help centre**
The H2 headings are slightly bigger than those on the deployed site.  Maybe this is an accessibility change?

**Comment by joepercival** [comment]
> **Cookies banner:** Clicking on “additional cookies” or “view cookies” is not loading the cookies page.
> **Cookies link in footer:** Clicking this just flicks the page to the header rather than show me the cookies? Note: it only exhibits this behaviour when the cookies banner is present.
These seem to be working for me, maybe a cache issue? There was an issue with javascript so it could potentially be fixed if you clear your cache

> **Home** “Useful resources” is a link now that returns 404. On the deployed website this isn’t a link.
This is because we are using a Card component here where we maybe shouldn't be, the design system changed the title on this component to be a link. Easy fix

> **Help centre** The H2 headings are slightly bigger than those on the deployed site. Maybe this is an accessibility change?
Not sure what this is at the minute but I'll take a look

**Comment by jhmjam** [comment]
It may be worth pulling main as well @joepercival as the cookies page would have changed now

**Comment by gibbardsteve** [comment]
> > **Cookies banner:** Clicking on “additional cookies” or “view cookies” is not loading the cookies page.
> > **Cookies link in footer:** Clicking this just flicks the page to the header rather than show me the cookies? Note: it only exhibits this behaviour when the cookies banner is present.
> 
> These seem to be working for me, maybe a cache issue? There was an issue with javascript so it could potentially be fixed if you clear your cache

Thanks - yep, it was a cache issue. Reloaded and it works ok.

File: terraform/cloudfront.tf
[the remark this answers is not in the document]
Comment (joepercival) [reply terraform/cloudfront.tf:1]: This is the javascript issue that Jason helped me with, I needed to update one of the javascript files which caused it to have a different hash to before, so the content security policy needed to be updated with that hash so the new script could be allowed to run

[the remark this answers is not in the document]
Comment (joepercival) [reply terraform/cloudfront.tf:1]: sml_builder/static/js/methods_catalogue.js

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d16l1gzj7635sf.cloudfront.net/
cloudfront_id=E3BVER8ZW4V1OV

**Review by gibbardsteve — APPROVED** [review]
Looks good

**Pull request merged**